### ISSUE TITLE
[Assistive tech and device support] Alert and status messages aren't properly exposed. (11.22.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#37066]
#### VFS product name
eGain

#### Point of Contact/Reviewers
Brian DeConinck (@briandeconinck) - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Alert and status messages aren't properly exposed.

### Issue Details
Error messages for the initial Name/Your Question form do not follow standard patterns. There are a few things I'm seeing:
- The messages are not programmatically associated with their inputs (eg. using `aria-describedby`)
- Messages are announced by screen readers when focus leaves an input only 
- The inputs themselves are not properly marked as required (visibly prior to user input, or programmatically with a `required` or `aria-required` attribute) 
- Inputs do not programmatically indicate when the input (or lack of input) is invalid, eg. using `aria-invalid`
- If a user inadvertently fails to enter something into one of the inputs, the "Start Chat" button simply doesn't work, with no visible or programmatic indicator that the button is disabled, no error message, and no way to return focus to the input that needs to be reviewed

### Link, screenshot or steps to recreate
1. Navigate to the [eGain chat tool](https://productionchat.vrm.vba.va.gov/system/templates/chat/Enrollment%20&%20Eligibility%20ACA/chat.html?entryPointId=1010&templateName=sunburst&languageCode=en&countryCode=US&ver=v11).
2. Activate a screen reader. (Recommend NVDA on Windows or VoiceOver on Mac.)
3. Press [Tab] to move through the page. Note that when focus is on the Name and Your Question inputs, neither are announced as required.
4. Leave the Name input blank, and press [Tab] to move to the Your Question input. Note that the error message "Enter name" is announced immediately after moving focus, with no context to indicate that it's an error. A screen reader user might misinterpret this as an instruction to enter their name in the field that now has focus, the Your Question input.
5. Press [Shift] + [Tab] to return to the Name field. Note that the screen reader does not announce the current error state (invalid input) or the error message while the field has focus.
6. Press [Tab] twice to move focus to the Start Chat button, leaving both the Name and Your Question inputs blank.
7. Press [Enter] to activate the Start Chat button. Note that nothing happens. The button is not activated, no error is announced, and no change in focus occurs. For a screen reader user who has inadvertently left an input blank, they receive no notification that they've made an error and may conclude that the tool just isn't working.

### VA.gov Experience Standard
[Category Number 11, Issue Number 22](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 4.1.3_AA

---

### Platform Recommendation
Follow established best practices for input validation and error messages. This includes: 
- Visually and programmatically indicate required fields prior to a user entering data. 
- When error messages are generated, associate them with their input programmatically using `aria-describedby`. 
- When an input or lack of input is not valid, indicate that programmatically using `aria-invalid`. 
- When the user attempts to press "Start Chat" with invalid data, provide an error message that receives keyboard focus and directs users to the right place to fix the error.

### VFS Team Tasks to Complete

- [ ] Comment on the ticket if there are questions or concerns
- [ ] Close the ticket when the issue has been resolved or validated by your Product Owner. If a team has additional questions or needs Platform help validating the issue, please comment in the ticket.
